include/GenSync/Syncs/BloomFilter.h

```diff
@@ -0,0 +1,231 @@
+/* This code is part of the GenSync project developed at Boston University.  Please see the README for use and references. */
+
+/*
+ * A Bloom filter is a space-efficient probabilistic data structure which tests
+ * whether an element is a member of a set. False positives are possible, but
+ * not false negatives. This data structure was developed by Burton Howard Bloom in 1970.
+ *
+ * Created by Anish Sinha on 12/4/23.
+ */
+
+#ifndef GENSYNCLIB_BLOOMFILTER_H
+#define GENSYNCLIB_BLOOMFILTER_H
+
+#include <vector>
+#include <utility>
+#include <string>
+#include <NTL/ZZ.h>
+#include <sstream>
+#include <GenSync/Aux/Auxiliary.h>
+#include <GenSync/Data/DataObject.h>
+
+using std::hash;
+using std::pair;
+using std::string;
+using std::stringstream;
+using std::vector;
+using namespace NTL;
+
+// Shorthand for the hash type
+typedef unsigned long int hash_t;

```

Comment:
Is this the right type for a hash output??  You could templatize the entire BloomFilter by this type, thereby allowing all kinds of hashes ... maybe.  This could also be a future issue to address.

Reply:
I believe based on the current implementation's use of the hash output, an unsigned long int would be ideal. However, templatizing the BloomFilter class by the hash type would be a good feature to have. I will write a separate issue/branch for this which will be addressed in the future.

Reply:
The problem is that the bit-length of an unsigned long int is system-dependent.  It might be better to have a fixed length version (e.g. uint64_t, or the like) that will be platform independent.  This is especially important when syncing different machines.